src/app/pages/psbt-request/use-psbt-request.tsx

```diff
@@ -1,68 +1,83 @@
 import { useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 import { bytesToHex, hexToBytes } from '@noble/hashes/utils';
+import * as btc from '@scure/btc-signer';
 
 import { finalizePsbt } from '@shared/actions/finalize-psbt';
-import { logger } from '@shared/logger';
-import { ensureArray, isUndefined } from '@shared/utils';
+import { RouteUrls } from '@shared/route-urls';
+import { ensureArray, undefinedIfLengthZero } from '@shared/utils';
 
 import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
 import { getPsbtPayloadFromToken } from '@app/common/psbt/requests';
+import { usePsbtRequestSearchParams } from '@app/common/psbt/use-psbt-request-params';
 import { usePsbtSigner } from '@app/features/psbt-signer/hooks/use-psbt-signer';
-import { usePsbtRequestSearchParams } from '@app/pages/psbt-request/psbt-request.hooks';
 
 export function usePsbtRequest() {
-  const { requestToken, tabId } = usePsbtRequestSearchParams();
   const [isLoading, setIsLoading] = useState(false);
-  const { signPsbt, signPsbtAtIndex, getDecodedPsbt, getPsbtAsTransaction } = usePsbtSigner();
+  const navigate = useNavigate();
+  const { origin, requestToken, tabId } = usePsbtRequestSearchParams();
+
+  const { signPsbt, getDecodedPsbt, getPsbtAsTransaction } = usePsbtSigner();
+
   const analytics = useAnalytics();
+
   return useMemo(() => {
     if (!requestToken) throw new Error('Cannot decode psbt without request token');
+
     const payload = getPsbtPayloadFromToken(requestToken);
     const payloadTxBytes = hexToBytes(payload.hex);
     const appName = payload?.appDetails?.name;
+    const allowedSighash = undefinedIfLengthZero(
+      payload?.allowedSighash?.map(h => Number(h)) as btc.SignatureHash[]
+    );
+    const indexesToSign = undefinedIfLengthZero(
+      ensureArray(payload?.signAtIndex)?.map(h => Number(h))
+    );
 
     return {
       appName,
+      allowedSighash,
+      indexesToSign,
       isLoading,
-      psbtPayload: payload,
       getDecodedPsbt,
+      origin,
+      psbtPayload: payload,
       get decodedPsbt() {
-        return getDecodedPsbt(payload.hex);
+        try {
+          return getDecodedPsbt(payload.hex);
+        } catch (e) {
+          return navigate(RouteUrls.RequestError, {
+            state: { message: e instanceof Error ? e.message : '', title: 'Failed request' },
+          });
```

Comment:
I wouldn't put side effects like this in a getter. Can we do a separate check somewhere to make sure we can decode the psbt?